nextstrain/cli/command/build.py

```diff
@@ -77,4 +98,94 @@ def run(opts):
 
         return 1
 
-    return runner.run(opts, working_volume = opts.build)
+    # Automatically pass thru appropriate resource options to Snakemake to
+    # avoid the user having to repeat themselves (once for us, once for
+    # snakemake).
+    if opts.exec == "snakemake":
+        snakemake_opts = parse_snakemake_args(opts.extra_exec_args)
+
+        if opts.cpus:
+            if not snakemake_opts["--cores"]:
+                opts.extra_exec_args += ["--cores=%d" % opts.cpus]
+            else:
+                warn(dedent("""
+                    Warning: The explicit %s option passed to Snakemake prevents
+                    the Nextstrain CLI from automatically providing one based on its
+                    --cpus option.  This may or may not be what you expect.
+                    """ % (snakemake_opts["--cores"][0],)))
+
+        if opts.memory:
+            if not snakemake_opts["--resources"]:
+                # Named MB but is really MiB, so convert our count of bytes to MiB
+                opts.extra_exec_args += ["--resources=mem_mb=%d" % (opts.memory // 1024**2)]
+            else:
+                # XXX TODO: Support parsing of --resources to see if "mem_mb" is
+                # provided.  If it's not, we could add our own "mem_mb" constraint
+                # alongside the other values of --resources.  Punting on this
+                # because it's not as simple as appending an additional argument.
+                # So for now, if folks are specifying their own --resources,
+                # they'll also need to explicitly provide "mem_mb", which may mean
+                # repeating a previous --memory argument they provided us.
+                #   -trs, 20 May 2020
+                warn(dedent("""
+                    Warning: The explicit %s option passed to Snakemake prevents
+                    the Nextstrain CLI from automatically providing a "mem_mb" resource
+                    based on its --memory option.  This may or may not be what you expect.
+                    """ % (snakemake_opts["--resources"][0],)))
+
```

Comment:
Could we utilize the fact that `opts.exec == "snakemake"` is the one "happy" conditional here to reduce the indentation of this code block?

Reply:
Hmm, good thought. I think that'd require repeating the `runner.run` call though?

```python
if opts.exec != "snakemake":
   return runner.run(opts, working_volume = opts.build, cpus = opts.cpus, memory = opts.memory)

# …bunch of arg manipulation here, then…

return runner.run(opts, working_volume = opts.build, cpus = opts.cpus, memory = opts.memory)
```

Not sure I like repeating that, as it seems easy to miss the two places when updating args.

Reply:
Oh, right. That would cause unnecessary duplication. I suppose there's no  good way around it.